### ISSUE TITLE
Handle all Ironic states

### ISSUE THREAD

**Issue by stbenjam** [openshift-metalkube/terraform-provider-ironic#17]
We should handle all  Ironic states, including all the possible failed states. 

e.g. `clean wait` wasn't handled.  That was fixed here - https://github.com/openshift-metalkube/terraform-provider-ironic/pull/16, need to handle the rest.

See: https://docs.openstack.org/ironic/pike/_images/states.svg

**Comment by dtantsur** [comment]
Also some handling is concerning. E.g. if `enroll` -> `manageable` transition fails, we go back to `enroll`. In this case the existing code will loop forever.

Similarly, the cleaning and deploying handling code does not account for errors, trying to redeploy/reclean instead (it won't work).

**Comment by dtantsur** [comment]
And a small correction: https://docs.openstack.org/ironic/stein/_images/states.svg is more up-to-date (pike is old, nearly out of support).

**Comment by stbenjam** [comment]
> Also some handling is concerning. E.g. if enroll -> manageable transition fails, we go back to enroll. In this case the existing code will loop forever.

How does one detect failure? Is `last_error` non-blank - is that the only way? It's odd this one doesn't have a failed state.

**Comment by dtantsur** [comment]
Yeah, this one is a bit weird. However, `enroll` + `last_error` should give you enough information. A node in `enroll` cannot get a non-empty `last_error` any other way.

**Comment by dhellmann** [comment]
Is this still an issue?

**Comment by stbenjam** [comment]
Yes. Enroll -> manageable failure isn’t handled yet at least.
